sql/wsrep_mysqld.cc

```diff
@@ -2285,6 +2292,124 @@ static void wsrep_NBO_begin_failed(THD *thd,
   unireg_abort(1);
 }
 
+/*
+  This function decide whether a DB operation can continue when it is using a
+  read only schema.
+
+  If a schema is read only, then this function shall disallow TOI operations on
+  all database objects(tables, triggers, functions, views, procedures, events)
+  inside the read only schemas except ALTER SCHEMA <schema> READ ONLY = 0.
+
+  @param thd          Thread context.
+  @param schema_name  Name of schema to check.
+
+  @return false if the DB operation can continue, true if not.
+*/
+static bool handle_read_only_schema(THD *thd, const char *db_,
+                                    const TABLE_LIST *table_list) {
+  /*
+    Disallow TOI operations on database objects inside read only schemas before
+    the query is replicated as TOI.
+
+    If the schema is in read_only state, then the only change allowed is to:
+
+    - Turn off read_only, possibly along with other option changes.
+    - Keep read_only turned on, i.e., a no-op. In this case, other options may
+      not be changed in the same statement.
+
+     This means we fail if:
+
+     - query is not ALTER DB
+     - query is ALTER DB but
+          - HA_CREATE_USED_READ_ONLY is not set.
+          - Or if we have set other fields as well and set READ ONLY to true.
+
+  */
+  LEX *lex = thd->lex;
+  HA_CREATE_INFO *create_info = lex->create_info;
+  bool some_tables_in_read_only_schema{false};
```

Comment:
Let's  keep consistent variables initialization way across the file `bool my_bool = false`